pontoon/base/models.py

```diff
@@ -2870,7 +2870,10 @@ def update_latest_translation(self):
         locale = self.locale
         translatedresource = TranslatedResource.objects.get(resource=resource, locale=locale)
 
-        instances = [project, locale, translatedresource]
+        instances = [translatedresource, project]
+
+        if Project.objects.visible().filter(pk=project.pk).first():
```

Comment:
Is there a better way to do this in Django? I am not that familiar with the DB layer, but it seems to me there are three SELECTs to the database first (`TranslatedResource.objects.`, `Project.objects.` and `utils.get_object_or_none`) to decide whether to update the three tables or not.

Maybe it could be rewritten to something that issues just three UPDATE statements instead, with the proper WHERE clause to ensure the right rows are (not) updated.

Reply:
It's enough to check if the project isn't a system project here like this:
```
if not project.system_project:
```

That's also what we do in `adjust_all_stats()` to make sure we don't update stats on `/teams` when system projects change.

The `Project.visible()` method would also check if the project is available (i.e. has been synced for the first time), but we don't need that since in this case the `Translation.save()` method (which calls `update_latest_translation()`) would not be callable.

Reply:
Nice, that's one SELECT down.

Do you think I should also get rid of the `get` on `TranslatedResource` and `ProjectLocale`? There wouldn't be the for-loop, just `X.objects.filter(...).update(...)`.

Reply:
You should be able to save another DB request by getting rid of `TranslatedResource.objects.get()` if you pass `translatedresource` to `self.update_latest_translation()` in `save()`.

I'm not sure about the `ProjectLocale`.

You can use this patch to debug SQL performance and compare the number of DB calls:
https://gist.github.com/mathjazz/92d2f6a36c968e8c1f6496f5e5aa0318

Reply:
I have tried to do it via QuerySet update() API. `update_latest_translation` is sometimes called from a context, where the translated resource is not present, or at least not directly.